--- lzop-1.01/src/conf.h.lzo2	2003-04-28 07:00:02.000000000 +0900
+++ lzop-1.01/src/conf.h	2005-06-11 12:30:57.256569328 +0900
@@ -34,13 +34,13 @@
 #endif
 
 #define WITH_LZO
-#include <lzoconf.h>
+#include <lzo/lzoconf.h>
 #if !defined(LZO_VERSION)
 #  error "you need the LZO library"
 #elif (LZO_VERSION < 0x1040)
 #  error "please upgrade your LZO package"
 #elif (LZO_VERSION >= 0x1080)
-#  include <lzoutil.h>
+#  include <lzo/lzoutil.h>
 #endif
 
 /* workaround a cpp bug under hpux 10.20 */
@@ -69,14 +69,14 @@
 **************************************************************************/
 
 #undef USE_MALLOC
-#if defined(WITH_LZO)
+#if 0 // defined(WITH_LZO)
 #  define USE_MALLOC
 #  undef malloc
 #  undef free
 #  define malloc        lzo_malloc
 #  define free          lzo_free
-#else
-#  error
+// #else
+// #  error
 #endif
 
 
@@ -682,7 +682,7 @@
 
 #if defined(WITH_LZO)
 
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 
 #define USE_LZO1X_1
 #if !defined(LZO_999_UNSUPPORTED)
@@ -693,12 +693,12 @@
 #endif
 
 #if defined(LZO_USE_ASM_1)
-LZO_EXTERN_CDECL(int)
+LZO_EXTERN(int)
 lzo1x_decompress_asm_fast
                         ( const lzo_byte *src, lzo_uint  src_len,
                                 lzo_byte *dst, lzo_uint *dst_len,
                                 lzo_voidp wrkmem /* NOT USED */ );
-LZO_EXTERN_CDECL(int)
+LZO_EXTERN(int)
 lzo1x_decompress_asm_fast_safe
                         ( const lzo_byte *src, lzo_uint  src_len,
                                 lzo_byte *dst, lzo_uint *dst_len,
@@ -706,12 +706,12 @@
 #  define lzo1x_decompress          lzo1x_decompress_asm_fast
 #  define lzo1x_decompress_safe     lzo1x_decompress_asm_fast_safe
 #elif defined(LZO_USE_ASM_2)
-LZO_EXTERN_CDECL(int)
+LZO_EXTERN(int)
 _lzo1x_decompress_asm_fast
                         ( const lzo_byte *src, lzo_uint  src_len,
                                 lzo_byte *dst, lzo_uint *dst_len,
                                 lzo_voidp wrkmem /* NOT USED */ );
-LZO_EXTERN_CDECL(int)
+LZO_EXTERN(int)
 _lzo1x_decompress_asm_fast_safe
                         ( const lzo_byte *src, lzo_uint  src_len,
                                 lzo_byte *dst, lzo_uint *dst_len,
--- lzop-1.01/configure.ac.lzo2	2003-04-28 07:00:02.000000000 +0900
+++ lzop-1.01/configure.ac	2005-06-11 12:29:09.075015424 +0900
@@ -65,12 +65,12 @@
 # // Checks for LZO library
 # ************************************************************************/
 
-AC_CHECK_HEADERS(lzoconf.h lzo1x.h,,AC_MSG_ERROR([LZO header file not found. Please check your installation or set the environment variable \`CPPFLAGS'.]))
+AC_CHECK_HEADERS(lzo/lzoconf.h lzo/lzo1x.h,,AC_MSG_ERROR([LZO header file not found. Please check your installation or set the environment variable \`CPPFLAGS'.]))
 
-AC_CHECK_LIB(lzo,__lzo_init2,,AC_MSG_ERROR([LZO library not found. Please check your installation or set the environment variable \`LDFLAGS'.]))
+AC_CHECK_LIB(lzo2,__lzo_init_v2,,AC_MSG_ERROR([LZO library not found. Please check your installation or set the environment variable \`LDFLAGS'.]))
 
-AC_CHECK_LIB(lzo,lzo1x_decompress_asm_fast,AC_DEFINE(LZO_USE_ASM_1,1,[Use asm.]))
-AC_CHECK_LIB(lzo,_lzo1x_decompress_asm_fast,AC_DEFINE(LZO_USE_ASM_2,1,[Use asm.]))
+AC_CHECK_LIB(lzo2,lzo1x_decompress_asm_fast,AC_DEFINE(LZO_USE_ASM_1,1,[Use asm.]))
+AC_CHECK_LIB(lzo2,_lzo1x_decompress_asm_fast,AC_DEFINE(LZO_USE_ASM_2,1,[Use asm.]))
 
 
 # /***********************************************************************
