--- theorarm-merge-branch/lib/arm/ARMstate.c	2010-05-12 17:06:37.000000000 -0700
+++ theorarm-merge-branch.save2/lib/arm/ARMstate.c	2010-08-03 15:54:57.000000000 -0700
@@ -26,7 +26,7 @@
                            const ptrdiff_t *_fragis,
                            const ptrdiff_t *frag_buf_offs);
 
-static void oc_state_frag_copy_list_arm(const oc_theora_state *_state,
+void oc_state_frag_copy_list_arm(const oc_theora_state *_state,
  const ptrdiff_t *_fragis,ptrdiff_t _nfragis,
  int _dst_frame,int _src_frame,int _pli){
   const ptrdiff_t     *frag_buf_offs;
@@ -43,7 +43,7 @@
                    frag_buf_offs);
 }
 
-void oc_state_loop_filter_frag_rows_arm(const oc_theora_state *_state,signed char*_bv,
+void oc_state_loop_filter_frag_rows_arm(const oc_theora_state *_state,int _bv[256],
  int _refi,int _pli,int _fragy0,int _fragy_end){
   const oc_fragment_plane *fplane;
   const oc_fragment       *frags;
@@ -83,7 +83,7 @@
                                        nhfrags);
 }
 
-static void oc_state_frag_recon_arm(const oc_theora_state *_state,ptrdiff_t _fragi,
+void oc_state_frag_recon_arm(const oc_theora_state *_state,ptrdiff_t _fragi,
  int _pli,ogg_int16_t _dct_coeffs[64],int _last_zzi,ogg_uint16_t _dc_quant){
   unsigned char *dst;
   ptrdiff_t      frag_buf_off;
@@ -128,15 +128,15 @@
 
 
 void oc_state_vtable_init_arm(oc_theora_state *_state){
-  _state->opt_vtable.frag_copy=NULL;
+  _state->opt_vtable.frag_copy=oc_frag_copy_c;
   _state->opt_vtable.frag_recon_intra=oc_frag_recon_intra_arm;
   _state->opt_vtable.frag_recon_inter=oc_frag_recon_inter_arm;
   _state->opt_vtable.frag_recon_inter2=oc_frag_recon_inter2_arm;
   _state->opt_vtable.idct8x8=oc_idct8x8_arm;
   _state->opt_vtable.state_frag_recon=oc_state_frag_recon_arm;
-  _state->opt_vtable.state_frag_copy_list=oc_state_frag_copy_list_arm;
+  _state->opt_vtable.state_frag_copy_list=oc_state_frag_copy_list_c;
   _state->opt_vtable.state_loop_filter_frag_rows=
-   oc_state_loop_filter_frag_rows_arm;
+   oc_state_loop_filter_frag_rows_c;
   _state->opt_vtable.restore_fpu=oc_restore_fpu_arm;
   _state->opt_data.dct_fzig_zag=OC_FZIG_ZAG;
 }
