--- orig-mtd-utils/flash_erase.c	2009-04-19 09:15:51.000000000 -0700
+++ mtd-utils/flash_erase.c	2009-04-19 09:29:47.000000000 -0700
@@ -54,6 +54,7 @@
 	{
 		erase_info_t erase;
 		region_info_t * r = &(reginfo[i]);
+		loff_t offset;
 
 		erase.start = start;
 		erase.length = r->erasesize;
@@ -70,13 +71,21 @@
 		printf("\rPerforming Flash Erase of length %u at offset 0x%x",
 				erase.length, erase.start);
 		fflush(stdout);
-		if(ioctl(Fd, MEMERASE, &erase) != 0)
+
+		offset = erase.start;
+		if (ioctl(Fd, MEMGETBADBLOCK, &offset) > 0)
 		{
-			perror("\nMTD Erase failure");
-			close(Fd);
-			return 8;
+			printf ("\nSkipping bad block at 0x%08x\n", erase.start);
+		}
+		else
+		{
+			if(ioctl(Fd, MEMERASE, &erase) != 0)
+			{
+				perror("\nMTD Erase failure");
+				close(Fd);
+				return 8;
+			}
 		}
-
 
 		start += erase.length;
 		if(start >= (r->offset + r->numblocks*r->erasesize))
@@ -94,6 +103,7 @@
 int non_region_erase(int Fd, int start, int count, int unlock)
 {
 	mtd_info_t meminfo;
+	loff_t offset;
 
 	if (ioctl(Fd,MEMGETINFO,&meminfo) == 0)
 	{
@@ -120,11 +130,19 @@
 				}
 			}
 
-			if (ioctl(Fd,MEMERASE,&erase) != 0)
+			offset = erase.start;
+			if (ioctl(Fd, MEMGETBADBLOCK, &offset) > 0)
 			{
-				perror("\nMTD Erase failure");
-				close(Fd);
-				return 8;
+				printf ("\nSkipping bad block at 0x%08x\n", erase.start);
+			}
+			else
+			{
+				if (ioctl(Fd,MEMERASE,&erase) != 0)
+				{
+					perror("\nMTD Erase failure");
+					close(Fd);
+					return 8;
+				}
 			}
 			erase.start += meminfo.erasesize;
 		}
