--- mtd-utils/flashcp.c	2009-02-03 19:03:08.000000000 -0800
+++ mtd-utils/flashcp.c.new	2009-09-03 12:55:30.000000000 -0700
@@ -99,6 +99,8 @@
 			"   -v | --verbose   Show progress reports\n"
 			"   <filename>       File which you want to copy to flash\n"
 			"   <device>         Flash device to write to (e.g. /dev/mtd0, /dev/mtd1, etc.)\n"
+			"\n"
+			" Note that if file is STDIN ('-') then the device is not verified\n"
 			"\n",
 			progname,progname);
 
@@ -109,6 +111,9 @@
 {
 	int fd;
 
+	if (!strcmp(pathname, "-"))	// handle '-' as STDIN
+		return STDIN_FILENO;
+
 	fd = open (pathname,flags);
 	if (fd < 0)
 	{
@@ -126,12 +131,12 @@
 	return (fd);
 }
 
-static void safe_read (int fd,const char *filename,void *buf,size_t count,bool verbose)
+ssize_t safe_read (int fd,const char *filename,void *buf,ssize_t count,bool verbose)
 {
 	ssize_t result;
 
 	result = read (fd,buf,count);
-	if (count != result)
+	if (count != result && fd != STDIN_FILENO)
 	{
 		if (verbose) log_printf (LOG_NORMAL,"\n");
 		if (result < 0)
@@ -142,6 +147,7 @@
 		log_printf (LOG_ERROR,"Short read count returned while reading from %s\n",filename);
 		exit (EXIT_FAILURE);
 	}
+	return(result);
 }
 
 static void safe_rewind (int fd,const char *filename)
@@ -168,7 +174,7 @@
 	const char *progname,*filename = NULL,*device = NULL;
 	int i,flags = FLAG_NONE;
 	ssize_t result;
-	size_t size,written;
+	size_t size,written,bytes_read;
 	struct mtd_info_user mtd;
 	struct erase_info_user erase;
 	struct stat filestat;
@@ -235,14 +241,23 @@
 
 	/* get some info about the file we want to copy */
 	fil_fd = safe_open (filename,O_RDONLY);
-	if (fstat (fil_fd,&filestat) < 0)
+	if (fil_fd == STDIN_FILENO)	// reading from stdin, input size equals mtd size
+		filestat.st_size = mtd.size;
+
+	else if (fstat (fil_fd,&filestat) < 0)
 	{
 		log_printf (LOG_ERROR,"While trying to get the file status of %s: %m\n",filename);
 		exit (EXIT_FAILURE);
 	}
+	if (filestat.st_size == 0)
+	{
+		// reading from magic file (like another mtd)...
+		// input size equals mtd size
+		filestat.st_size = mtd.size;
+	}
 
 	/* does it fit into the device/partition? */
-	if (filestat.st_size > mtd.size)
+	if ((unsigned)filestat.st_size > mtd.size)
 	{
 		log_printf (LOG_ERROR,"%s won't fit into %s!\n",filename,device);
 		exit (EXIT_FAILURE);
@@ -252,8 +267,6 @@
 	 * erase enough blocks so that we can write the file *
 	 *****************************************************/
 
-#warning "Check for smaller erase regions"
-
 	erase.start = 0;
 	erase.length = filestat.st_size & ~(mtd.erasesize - 1);
 	if (filestat.st_size % mtd.erasesize) erase.length += mtd.erasesize;
@@ -295,10 +308,17 @@
 	 * write the entire file to flash *
 	 **********************************/
 
-	if (flags & FLAG_VERBOSE) log_printf (LOG_NORMAL,"Writing data: 0k/%luk (0%%)",KB (filestat.st_size));
+	if (flags & FLAG_VERBOSE) {
+		if (fil_fd == STDIN_FILENO) {
+			log_printf (LOG_NORMAL,"Writing STDIN to flash\n");
+		} else {
+			log_printf (LOG_NORMAL,"Writing data: 0k/%luk (0%%)",KB (filestat.st_size));
+		}
+	}
 	size = filestat.st_size;
 	i = BUFSIZE;
 	written = 0;
+	bytes_read = 0;
 	while (size)
 	{
 		if (size < BUFSIZE) i = size;
@@ -309,36 +329,51 @@
 					PERCENTAGE (written + i,filestat.st_size));
 
 		/* read from filename */
-		safe_read (fil_fd,filename,src,i,flags & FLAG_VERBOSE);
+		bytes_read = safe_read (fil_fd,filename,src,i,flags & FLAG_VERBOSE);
+
+		/* if result is zero and file is STDIN then we're finished */
+		if (bytes_read == 0 && fil_fd == STDIN_FILENO)
+			break;
 
 		/* write to device */
-		result = write (dev_fd,src,i);
-		if (i != result)
+		result = write (dev_fd,src,bytes_read);
+		if (bytes_read != result)
 		{
 			if (flags & FLAG_VERBOSE) log_printf (LOG_NORMAL,"\n");
 			if (result < 0)
 			{
 				log_printf (LOG_ERROR,
 						"While writing data to 0x%.8x-0x%.8x on %s: %m\n",
-						written,written + i,device);
+						written,written + bytes_read,device);
 				exit (EXIT_FAILURE);
 			}
 			log_printf (LOG_ERROR,
 					"Short write count returned while writing to x%.8x-0x%.8x on %s: %d/%lu bytes written to flash\n",
-					written,written + i,device,written + result,filestat.st_size);
+					written,written + bytes_read,device,written + bytes_read,filestat.st_size);
 			exit (EXIT_FAILURE);
 		}
 
-		written += i;
-		size -= i;
+		written += bytes_read;
+		size -= bytes_read;
 	}
-	if (flags & FLAG_VERBOSE)
-		log_printf (LOG_NORMAL,
+
+	if (flags & FLAG_VERBOSE) {
+		if (fil_fd == STDIN_FILENO) {
+			log_printf (LOG_NORMAL, "\rWrote: %luk bytes             \n",
+				KB (written));
+			DEBUG("Wrote %d / %luk bytes\n", written);
+		} else {
+			log_printf (LOG_NORMAL,
 				"\rWriting data: %luk/%luk (100%%)\n",
-				KB (filestat.st_size),
+				KB (written),
 				KB (filestat.st_size));
-	DEBUG("Wrote %d / %luk bytes\n",written,filestat.st_size);
+			DEBUG("Wrote %d / %luk bytes\n",written,filestat.st_size);
+		}
+	}
 
+	if (fil_fd == STDIN_FILENO)
+		exit (EXIT_SUCCESS);
+		
 	/**********************************
 	 * verify that flash == file data *
 	 **********************************/
