diff -Naur mtd-utils-orig/nandtest.c mtd-utils/nandtest.c
--- mtd-utils-orig/nandtest.c	2009-02-03 19:03:08.000000000 -0800
+++ mtd-utils/nandtest.c	2010-05-06 09:34:26.194641571 -0700
@@ -100,7 +100,7 @@
 	}
 	if (newstats.failed > oldstats.failed) {
 		printf("\nECC failed at %08x\n", (unsigned) ofs);
-		oldstats.corrected = newstats.corrected;
+		oldstats.failed = newstats.failed;
 	}
 	if (len < meminfo.erasesize)
 		exit(1);
