diff -rupN emerald-boot.mod/copyNand.c emerald-boot/copyNand.c
--- emerald-boot.mod/copyNand.c	2011-04-18 15:14:38.000000000 -0700
+++ emerald-boot/copyNand.c	2011-07-03 01:27:03.000000000 -0700
@@ -23,13 +23,14 @@ u32 GetCopySize();  // Returns the value
 
 void nand_bootstrap()
 {
+#ifdef SELF_BOOTSTRAP
 	const u32 page_size = 2048;
 	const u32 page_shift = 11;
 
 	/* Note: Turn off this option if you are testing emerald-boot by
 	 *       forcing it into address 0 via u-boot or some other means like
 	 *       JTAG. */
-#ifdef SELF_BOOTSTRAP
+
 	/* load the rest of the boot module (hardware loaded 2K) */
 	// Call nand_read_bootstrap() to read (u32)_copy_end - BOOTSTRAP_SIZE
 	// (rounded up to the nearest multiple of the page size) bytes
diff -rupN emerald-boot.mod/install.sh emerald-boot/install.sh
--- emerald-boot.mod/install.sh	2011-02-28 16:33:24.000000000 -0800
+++ emerald-boot/install.sh	2011-07-03 01:27:03.000000000 -0700
@@ -2,6 +2,19 @@
 
 # This script builds a emerald-boot image.
 
+BUILD_REVISION=0001
+
+# Add boot type suffix to output bin
+if [ $MICRO_BOOT -eq 1 ]; then
+        BOOT_TYPE=UART
+else
+        BOOT_TYPE=NOR
+fi
+# output file name
+LEX_TARGET=emerald-boot_$BOOT_TYPE
+LPAD_TARGET=madrid-boot_$BOOT_TYPE
+
+
 set -e
 
 . $PROJECT_PATH/scripts/functions
@@ -20,11 +33,11 @@ if [ "$CLEAN" == "1" ]; then
 	make clean
 fi
 
-SDCARD=$SDCARD SDCARD_DEBUG=$SDCARD_DEBUG UBOOT_SUPPORT=$UBOOTLOADERS KERNELDIR=$KERNELDIR CROSS_COMPILE=$CROSS_COMPILE make 
+SDCARD=$SDCARD SDCARD_DEBUG=$SDCARD_DEBUG UBOOT_SUPPORT=$UBOOTLOADERS KERNELDIR=$KERNELDIR CROSS_COMPILE=$CROSS_COMPILE LX_TARGET=$LEX_TARGET LPD_TARGET=$LPAD_TARGET make 
 
 VERSION=`grep "CURRENT_MAJOR_VERSION=" ../packages/version/install.sh | cut -d = -f 2`
-BUILD=`svn info $PROJECT_PATH | grep Revision | awk '{print $2}'`
-for b in emerald-boot.bin madrid-boot.bin; do
+BUILD=$BUILD_REVISION
+for b in $LEX_TARGET.bin $LPAD_TARGET.bin; do
 	echo "" >> $b
 	echo "VERSION:$VERSION.$BUILD" >> $b
 	echo "" >> $b
diff -rupN emerald-boot.mod/main.c emerald-boot/main.c
--- emerald-boot.mod/main.c	2011-05-05 08:07:02.000000000 -0700
+++ emerald-boot/main.c	2011-07-27 14:15:18.000000000 -0700
@@ -849,14 +849,20 @@ void boot_nand(void)
 }
 
 
-void boot_sd1(void)
+void boot_sd(u32 SD_BASE)
 {
 	struct disk_partition partitions[4];
 	global_var * gptr = get_global();
 	FW_DEBUG_0_ON;
-	gptr->found_sd1 = disk_load_mbr(SDHC1_BASE, io_buffer, partitions) == 0;
-	if (gptr->found_sd1) {
-		serio_puts("SD1 found\n");
+	if ( SD_BASE == SDHC0_BASE ) {
+		gptr->found_sd0 = disk_load_mbr(SDHC0_BASE, io_buffer, partitions) == 0;
+	}
+	else {
+		gptr->found_sd1 = disk_load_mbr(SDHC1_BASE, io_buffer, partitions) == 0;
+
+	}
+	if ( gptr->found_sd0 || gptr->found_sd1 ) {
+		serio_puts("SD found\n");
 		/* construct the command line */
 		strcpy(gptr->cmdline_buffer, CMDLINE_BASE);
 		strcat(gptr->cmdline_buffer, gptr->ram_cmdline);
@@ -868,18 +874,15 @@ void boot_sd1(void)
 		strcat(gptr->cmdline_buffer, CMDLINE_RFS_SD1);
 		/* Read in */
 		//FW_DEBUG_1_ON;	// disk_load_mbr() costs 350msec
-		gptr->image = load_sd(SDHC1_BASE, partitions, io_buffer,
+		gptr->image = load_sd(SD_BASE, partitions, io_buffer,
 				gptr->params_buffer);
 		if (gptr->image) {
-			serio_puts ("SD1 kernel passes CBF\n");
+			serio_puts ("SD kernel passes CBF\n");
 		}
 		else {
-			serio_puts ("SD1 kernel fails CBF\n");
+			serio_puts ("SD kernel fails CBF\n");
 		}
 	}
-	else {
-		serio_puts("No SD1 found\n");
-	}
 }
 
 
@@ -1050,25 +1053,37 @@ void main(void)
 	 * Look for a bootable media device and
 	 * if found, bring in a kernel checksum
 	 */
-		// If booted from nand or if not a madrid board, search_for_nand
-	if (    IS_SET(REG32(MCU_S_BASE+NFCONTROL),NFBOOTENB)
-		|| !isMadrid(gptr->board_id))
+
+	// Try and boot off SD0 if fails, proceed normally
+
+	serio_puts("Trying to boot from SD0\n");
+	boot_sd(SDHC0_BASE);
+
+	if ( gptr->image == 0 )
 	{
-		if ( !gptr->image ) {
-			search_for_nand();
-			if ( gptr->found_nand )
-				boot_nand();
+		serio_puts("\nSD0 Not Found\n");
+		// If booted from nand or if not a madrid board, search_for_nand
+		if (    IS_SET(REG32(MCU_S_BASE+NFCONTROL),NFBOOTENB)
+			|| !isMadrid(gptr->board_id))
+		{
+			serio_puts("Trying to boot from NAND\n");
+			if ( !gptr->image ) {
+				search_for_nand();
+				if ( gptr->found_nand )
+					boot_nand();
+			}
 		}
-	}
-#if 0	// original
-	if ( !gptr->image ) 
-#else	
-	else	// new: call boot_sd1() only if (isMadrid() && not a NAND boot) 
-#endif
-	{
-		boot_sd1();
-	}
+	#if 0	// original
+		if ( !gptr->image ) 
+	#else	
+		else	// new: call boot_sd() only if (isMadrid() && not a NAND boot) 
+	#endif
+		{
+			serio_puts("Trying to boot from SD1\n");
+			boot_sd(SDHC1_BASE);
 
+		}
+	}
 	/* Nothing found? */
 	if (gptr->image == 0)
 	{
diff -rupN emerald-boot.mod/Makefile emerald-boot/Makefile
--- emerald-boot.mod/Makefile	2011-04-18 15:14:38.000000000 -0700
+++ emerald-boot/Makefile	2011-07-03 01:27:03.000000000 -0700
@@ -1,6 +1,6 @@
 CROSS_COMPILE ?= arm-linux-uclibcgnueabi-
 OPT=-Os
-CFLAGS = -ggdb -std=gnu99 -mcpu=arm926ej-s -mlittle-endian -msoft-float -ffreestanding $(OPT) -Wall -Werror -pedantic -Wno-variadic-macros -D_LF1000_BOOTLOADER -DSELF_BOOTSTRAP -fPIC -fno-builtin -I./include
+CFLAGS = -ggdb -std=gnu99 -mcpu=arm926ej-s -mlittle-endian -msoft-float -ffreestanding $(OPT) -Wall -Werror -pedantic -Wno-variadic-macros -D_LF1000_BOOTLOADER -fPIC -fno-builtin -I./include
 LDFLAGS = -static --architecture armv5te -EL -M -Map boot.map -e StartUp -T emerald-boot.lds
 
 # nand_utils.o and nand_boot.o must be in the first 2K.
@@ -26,8 +26,8 @@ LPAD_SCREENS=$(GENERIC_SCREENS:%=Lpad-%)
 # HNS_SIZE = `cat $(SCREENS_PATH)/screens/FIRST.32.rle | wc -c` 
 HNS_SIZE = `cat $(SCREENS_PATH)/screens/Madrid-Boot-logo.32.rle | wc -c` 
 
-LEX_TARGET = emerald-boot
-LPAD_TARGET = madrid-boot
+LEX_TARGET = $(LX_TARGET)
+LPAD_TARGET = $(LPD_TARGET)
 
 LEX_OBJS = $(OBJS) usbDescriptors-emerald.o lex_screens.o rle.o mlc.o
 LPAD_OBJS = $(OBJS) usbDescriptors-madrid.o lpad_screens.o rle.o mlc.o
@@ -35,6 +35,15 @@ LPAD_OBJS = $(OBJS) usbDescriptors-madri
 LEX_RES = -DFB_VIRTUAL_XRES=320 -DFB_VIRTUAL_YRES=240
 LPAD_RES = -DFB_VIRTUAL_XRES=480 -DFB_VIRTUAL_YRES=272
 
+# Boot using Micromon, turn bootstrap off
+ifneq ($(MICRO_BOOT),)
+CFLAGS += -DMICRO_BOOT
+#BOOT_TYPE = UART
+else
+CFLAGS += -DSELF_BOOTSTRAP
+#BOOT_TYPE = NOR
+endif
+
 
 ## Enable debugging for now...
 # CFLAGS += -DMMC_DEBUG -DDEBUG
diff -rupN emerald-boot.mod/startup.S emerald-boot/startup.S
--- emerald-boot.mod/startup.S	2011-04-18 15:14:38.000000000 -0700
+++ emerald-boot/startup.S	2011-07-03 01:27:03.000000000 -0700
@@ -316,6 +316,16 @@ DisableInterrupts:
 
     /* After power-on or reset, execution starts here */
 ResetHandler:
+
+         #ifdef MICRO_BOOT
+           // Sets NFCONTROL bit NFBOOTENB, Doesn't seem to get set
+           //  when put into UART/NAND boot mode, so must manually do it
+           ldr r3, =0xC0015874   // Loads NFCONTROL's address into R3
+           ldr r2, [r3]          // Fetch NFCONTROL's value
+           orr r2, r2, #0x020   // Set Bit5
+           str r2, [r3]          // Store the value back to NFCONTROL
+        #endif
+
 	/* Enable I-Cache, copied from VTK/Bootloader/NandBoot/startup.s */
 	MRC     p15,0,r0,c1,c0,0
 	ORR     r0,r0,#(1<<12)
